src/transformer/utils/get_module_name.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace src\transformer\utils;
+
+function get_module_name(array $config, $moduletype, $module) {
+    $name = '';
+    if (property_exists($module, 'name')) {
+        $name = $module->name;
+    } else if (property_exists($moduletype, 'name')) {
```

Comment:
I'm not sure if that will work because I think the module type is usually a string at this point.

Reply:
Yes it is a string, that is what we need is a string. The issue is the $module->name does not always exist especially for a quiz, because quizzes do not have a name. So that is why I revert to using $moduletype->name when module does not have a name. It was the fix that allowed mod_quiz events to be properly sent. It was being rejected before with element can not be 'null'

Reply:
Yeah sorry, I'm saying that the `$moduletype` variable is a string not an object, so the name property will never exist. I think you need to change `$moduletype->name` to just `$moduletype` right?

Reply:
You are correct, I forgot to change that out after removing the temp fix I used earlier. Sorry about that.

Reply:
No worries, thanks for checking that Lee 👍